GitTfs/Core/GitTfsRemote.cs

```diff
@@ -168,6 +168,18 @@ private void InitHistory()
                 else
                 {
                     MaxChangesetId = 0;
+                    //Manage the special case where a .gitignore has been commited
```

Comment:
Can the special case be managed by setting `MaxCommitHash` directly, rather than implicitly figuring out that it needs to? Like maybe `CommitGitIgnore` could call `InitHistory` and then set `MaxCommitHash` manually? Or `InitHistory` could take an optional initial value for max commit hash?